.github/workflows/build_rpm.yaml

```
@@ -0,0 +1,176 @@
+
+      - name: Build RPM package
+        id: build-rpm-package
+        env:
+          build_dir: "$(pwd)/build"
+        run: |
+          export PATH=$PATH:$HOME/.cargo/bin/
+          mkdir -p ${{ env.build_dir }}/SOURCES
+          tar --exclude=".[^/]*" --exclude="./build" --transform 's|.|alumet|' -zcvf ${{ env.build_dir }}/SOURCES/alumet-agent.tar.gz .
+          rpmbuild -ba -vv --define "_topdir ${{ env.build_dir }}" --define "release ${{ inputs.release-version }}" --define "osr ${{ matrix.os.name }}" --define "version ${{ inputs.build-version }}" --define "arch ${{ inputs.target-architecture }}" packaging/rpm/specs/alumet.spec
+
+      - name: Upload artifact alumet-agent
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.name }}-${{ inputs.build-version }}-${{ inputs.release-version }}.${{ matrix.os.name }}.${{ inputs.target-architecture }}.rpm
+          path: |
+            ./build/RPMS/${{ inputs.target-architecture }}/${{ env.name }}-${{ inputs.build-version }}-${{ inputs.release-version }}.${{ matrix.os.name }}.${{ inputs.target-architecture }}.rpm
+
+  testing:
+    runs-on: ubuntu-latest
+    needs:
+      - packaging
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - {name: fc40, base-image: fedora:40}
+          - {name: fc41, base-image: fedora:41}
+          - {name: fc42, base-image: fedora:42}
+          - {name: ubi8.3, base-image: registry.access.redhat.com/ubi8/ubi:8.3}
+          - {name: ubi9.5, base-image: registry.access.redhat.com/ubi9/ubi:9.5}
+    container:
+      image: ${{ matrix.os.base-image }}
+      options: --cap-add=SYS_NICE --cap-add=PERFMON
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          repository: 'alumet-dev/packaging'
+          fetch-depth: 1
+          path: 'packaging'
+          ref: 'main'
+
+      - name: Download RPM artifact
+        id: download_rpm
+        uses: actions/download-artifact@v4
+        with:
+          pattern: '**/*${{ matrix.os.name }}*.rpm'
+          path: ./rpm
+          merge-multiple: true
+
+      - name: Install RPM on Fedora
+        if: contains(matrix.os.base-image, 'fedora')
+        run: |
+          dnf install -y systemd
+
+      - name: Install RPMs
+        run: |
+          dnf install -y ${{ steps.download_rpm.outputs.download-path }}/*.rpm
+
+      - name: Install Goss
+        run: |
+          curl -fsSL https://goss.rocks/install | sh
+
+      - name: Run Goss tests
+        run: |
+          set -o pipefail
+          goss -g packaging/.github/goss_validate.yaml validate --format documentation | tee goss-results-rpm_${{ matrix.os.name }}.txt >> $GITHUB_STEP_SUMMARY
+
+      - name: Upload artifact alumet-agent
+        uses: actions/upload-artifact@v4
+        with:
+          name: goss-results-rpm_${{ matrix.os.name }}.txt
+          path: |
+            ./goss-results-rpm_${{ matrix.os.name }}.txt
```

This step won't be used in case of an error in the previous one.
use:
```suggestion
      - name: Upload artifact alumet-agent
         if: >-
           always() && 
           steps.goss-tests.outcome == 'failure'
        uses: actions/upload-artifact@v4
        with:
          name: goss-results-rpm_${{ matrix.os.name }}.txt
          path: |
            ./goss-results-rpm_${{ matrix.os.name }}.txt
```